core/payments/reservation/reservation_ledger.go

```diff
@@ -0,0 +1,117 @@
+package reservation
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/Layr-Labs/eigenda/core"
+)
+
+// Tracks usage of a single account reservation
+//
+// This struct is goroutine safe.
+type ReservationLedger struct {
+	config ReservationLedgerConfig
+
+	// synchronizes access to the underlying leaky bucket algorithm
+	lock sync.Mutex
+
+	// an instance of the algorithm which tracks reservation usage
+	leakyBucket *LeakyBucket
+}
+
+// Creates a new reservation ledger, which represents the reservation of a single user with a leaky bucket
+func NewReservationLedger(
+	config ReservationLedgerConfig,
```

Comment:
```suggestion
	config *ReservationLedgerConfig,
```

Reply:
Let's have the conversation here https://github.com/Layr-Labs/eigenda/pull/1873#discussion_r2277526418